stl/inc/format

```diff
@@ -343,72 +337,398 @@ public:
+        } else if constexpr (_Same_impl<_Ty, char> && _Same_impl<_CharType, wchar_t>) {
+            _Store_impl<_CharType, _Basic_format_arg_type::_Char_type>(_Arg_index, static_cast<wchar_t>(_Val));
+        } else if constexpr (signed_integral<_Ty> && sizeof(_Ty) <= sizeof(int)) {
+            _Store_impl<int, _Basic_format_arg_type::_Int_type>(_Arg_index, static_cast<int>(_Val));
+        } else if constexpr (unsigned_integral<_Ty> && sizeof(_Ty) <= sizeof(unsigned)) {
+            _Store_impl<unsigned, _Basic_format_arg_type::_UInt_type>(_Arg_index, static_cast<unsigned>(_Val));
+        } else if constexpr (signed_integral<_Ty> && sizeof(_Ty) <= sizeof(long long)) {
+            _Store_impl<long long, _Basic_format_arg_type::_Long_type>(_Arg_index, static_cast<long long>(_Val));
+        } else if constexpr (unsigned_integral<_Ty> && sizeof(_Ty) <= sizeof(unsigned long long)) {
+            _Store_impl<unsigned long long, _Basic_format_arg_type::_ULong_type>(
+                _Arg_index, static_cast<unsigned long long>(_Val));
+        } else {
+            _Store_impl<typename basic_format_arg<_Context>::handle, _Basic_format_arg_type::_Custom_type>(
+                _Arg_index, basic_format_arg<_Context>::handle(_Val));
+        }
+    }
+
+    void _Store(const size_t _Arg_index, float _Val) noexcept {
+        _Store_impl<float, _Basic_format_arg_type::_Float_type>(_Arg_index, _Val);
+    }
+
+    void _Store(const size_t _Arg_index, double _Val) noexcept {
+        _Store_impl<double, _Basic_format_arg_type::_Double_type>(_Arg_index, _Val);
+    }
+
+    void _Store(const size_t _Arg_index, long double _Val) noexcept {
+        _Store_impl<long double, _Basic_format_arg_type::_Long_double_type>(_Arg_index, _Val);
+    }
+
+    void _Store(const size_t _Arg_index, const _CharType* _Val) noexcept {
+        _Store_impl<const _CharType*, _Basic_format_arg_type::_CString_type>(_Arg_index, _Val);
+    }
+
+    template <class _Traits>
+    void _Store(const size_t _Arg_index, basic_string_view<_CharType, _Traits> _Val) noexcept {
+        _Store_impl<basic_string_view<_CharType>, _Basic_format_arg_type::_String_type>(
+            _Arg_index, basic_string_view<_CharType>{_Val});
+    }
+
+    template <class _Traits, class _Alloc>
+    void _Store(const size_t _Arg_index, basic_string<_CharType, _Traits, _Alloc> _Val) noexcept {
+        _Store_impl<basic_string_view<_CharType>, _Basic_format_arg_type::_String_type>(
+            _Arg_index, _Basic_format_arg_type::_String_type, basic_string_view<_CharType>{_Val.data(), _Val.size()});
+    }
+
+    void _Store(const size_t _Arg_index, nullptr_t) noexcept {
+        _Store_impl<const void*, _Basic_format_arg_type::_Pointer_type>(_Arg_index, static_cast<const void*>(nullptr));
+    }
+
+    // clang-format off
+    template <class _Ty>
+        requires is_void_v<_Ty>
+    void _Store(const size_t _Arg_index, const _Ty* p) noexcept {

```

Comment:
I think It is explicitly made to prevent volatile and discriminate between string literals.

That said I do not know why it was given in the standard like that https://eel.is/c++draft/format#arg-12